psalm-baseline.xml

```diff
@@ -1,3 +1,129 @@
+      <code>$filterValue</code>
+    </MixedAssignment>
+  </file>
+  <file src="src/Builder/Encoder/FieldPathEncoder.php">
+    <NoInterfaceProperties>
+      <code><![CDATA[$value->name]]></code>
+    </NoInterfaceProperties>
+  </file>
+  <file src="src/Builder/Encoder/OperatorEncoder.php">
+    <MixedAssignment>
+      <code>$result</code>
+      <code>$result[]</code>
+      <code>$val</code>
+      <code>$val</code>
+      <code>$val</code>
+      <code>$val</code>
+      <code>$val</code>
+    </MixedAssignment>
+    <UndefinedConstant>
+      <code>$value::ENCODE</code>
+    </UndefinedConstant>
+  </file>
+  <file src="src/Builder/Encoder/QueryEncoder.php">
+    <MixedAssignment>
+      <code>$subValue</code>
+      <code>$value</code>
+    </MixedAssignment>
+  </file>
+  <file src="src/Builder/Projection/ElemMatchOperator.php">
+    <MixedArgumentTypeCoercion>
+      <code>$query</code>
+    </MixedArgumentTypeCoercion>
+  </file>
+  <file src="src/Builder/Query.php">
+    <ArgumentTypeCoercion>
+      <code>$query</code>
+    </ArgumentTypeCoercion>
+  </file>
+  <file src="src/Builder/Query/ElemMatchOperator.php">
+    <MixedArgumentTypeCoercion>
+      <code>$query</code>
+    </MixedArgumentTypeCoercion>
+  </file>
+  <file src="src/Builder/Stage/AddFieldsStage.php">
+    <PropertyTypeCoercion>
+      <code>$expression</code>
+    </PropertyTypeCoercion>
+    <TooManyTemplateParams>
+      <code>stdClass</code>
+    </TooManyTemplateParams>
+  </file>
+  <file src="src/Builder/Stage/FacetStage.php">
+    <PropertyTypeCoercion>
+      <code>$facet</code>
+    </PropertyTypeCoercion>
+    <TooManyTemplateParams>
+      <code>stdClass</code>
+    </TooManyTemplateParams>
+  </file>
+  <file src="src/Builder/Stage/GeoNearStage.php">
+    <MixedArgumentTypeCoercion>
+      <code>$query</code>
+    </MixedArgumentTypeCoercion>
+  </file>
+  <file src="src/Builder/Stage/GraphLookupStage.php">
+    <MixedArgumentTypeCoercion>
+      <code>$restrictSearchWithMatch</code>
+    </MixedArgumentTypeCoercion>
+  </file>
+  <file src="src/Builder/Stage/GroupStage.php">
+    <PropertyTypeCoercion>
+      <code>$field</code>
+    </PropertyTypeCoercion>
+    <TooManyTemplateParams>
+      <code>stdClass</code>
+    </TooManyTemplateParams>
+  </file>
+  <file src="src/Builder/Stage/MatchStage.php">
+    <MixedArgumentTypeCoercion>
+      <code>$query</code>
+    </MixedArgumentTypeCoercion>
+  </file>
+  <file src="src/Builder/Stage/ProjectStage.php">
+    <PropertyTypeCoercion>
+      <code>$specification</code>
+    </PropertyTypeCoercion>
+    <TooManyTemplateParams>
+      <code>stdClass</code>
+    </TooManyTemplateParams>
+  </file>
+  <file src="src/Builder/Stage/SetStage.php">
+    <PropertyTypeCoercion>
+      <code>$field</code>
+    </PropertyTypeCoercion>
+    <TooManyTemplateParams>
+      <code>stdClass</code>
+    </TooManyTemplateParams>
+  </file>
+  <file src="src/Builder/Type/OutputWindow.php">
+    <DocblockTypeContradiction>
```

Comment:
This and the `RedundantConditionGivenDocblockType` checks are psalm telling us that the condition is impossible due to the docblock. However, since we're being extra cautious here and any user not using psalm is able to send any types they want, I think it's fair to keep this around unless we want to accept potential fatal errors when users abuse this.